spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/Log4J2LoggingSystem.java

```diff
@@ -155,10 +161,50 @@ public void beforeInitialize() {
 		if (isAlreadyInitialized(loggerContext)) {
 			return;
 		}
-		super.beforeInitialize();
+		if (!configureJdkLoggingBridgeHandler()) {
+			super.beforeInitialize();
+		}
 		loggerContext.getConfiguration().addFilter(FILTER);
 	}
 
+	private boolean configureJdkLoggingBridgeHandler() {
+		try {
+			if (isJulUsingASingleConsoleHandlerAtMost() && !isLog4jLogManagerInstalled()) {
+				if (isLog4jBridgeHandlerAvailable()) {
+					installLog4jBridgeHandler();
+					return true;
+				}
+			}
+		}
+		catch (Throwable ex) {
+			// Ignore. No java.util.logging bridge is installed.
+		}
+		return false;
+	}
+
+	private boolean isLog4jLogManagerInstalled() {
+		final String logManagerClassName = java.util.logging.LogManager.getLogManager().getClass().getName();
+		return LOG4J_LOG_MANAGER.equals(logManagerClassName);
+	}
+
+	private boolean isLog4jBridgeHandlerAvailable() {
+		return ClassUtils.isPresent(LOG4J_BRIDGE_HANDLER, getClassLoader());
+	}
+
+	private void installLog4jBridgeHandler() {
+		Log4jBridgeHandler.install(true, null, true);
```

Comment:
I think the first `true` here will cause a regression of the changes for https://github.com/spring-projects/spring-boot/issues/8933. Logic similar to `removeDefaultRootHandler` on `Slf4JLogingSystem ` needs to be used instead.

That makes me wonder if we need to take a slightly different approach overall, possibly moving the bridge handler-related logic out into a separate class that can then be used by the different logging systems. I'm not yet sure quite how that would look so I'll need to experiment a bit before I can make a more concrete suggestion.

Reply:
I added the logic used by the `Slf4JLoggingSystem`, which should solve the immediate problem.

The bridge logic should probably be refactor to separate classes. I started with this approach in commit [39ed0a1](https://github.com/spring-projects/spring-boot/pull/30003/commits/39ed0a12512402cf54dadbf7169a54af8febba8f), but then it looked as an overkill.

I think that especially the logic to decide whether to modify JUL's configuration or not should be expanded (a switch case on the name of `java.util.logging.LogManager` implementation?): all three logging systems must decide whether they can modify the root handler and how.